tests/rptest/tests/cloud_storage_scrubber_test.py

```diff
@@ -471,18 +546,53 @@ def gap_reported():
             backoff_sec=2,
             err_msg="Gap not reported")
 
-        if ntpr not in expected_anomalies:
-            expected_anomalies[ntpr] = Anomalies.make_empty(ntpr)
-
         expected_anomalies[ntpr].segment_metadata_anomalies.add(meta_anomaly)
 
+    def _assert_metrics_updated(self):
+        metrics = [
+            self.redpanda.metrics(n, MetricsEndpoint.PUBLIC_METRICS)
+            for n in self.redpanda.nodes
+        ]
+
+        missing_segments = 0
+        missing_spills = 0
+        offset_gaps = 0
+
+        for node_metrics in metrics:
+            for family in node_metrics:
+                for sample in family.samples:
```

Comment:
i'm not very familiar with redpanda metrics, but would it make sense to assert labels here to make sure we are aggregating metrics correctly? i.e. only per topic metrics are present, rather than per-partition.

Reply:
Good idea. I'll update the test with some validation of the labels.

Reply:
Done.